model/version.go

```diff
@@ -35,7 +35,10 @@ type Version struct {
 	RepoKind            string               `bson:"repo_kind" json:"repo_kind,omitempty"`
 	BuildVariants       []VersionBuildStatus `bson:"build_variants_status,omitempty" json:"build_variants_status,omitempty"`
 	PeriodicBuildID     string               `bson:"periodic_build_id,omitempty" json:"periodic_build_id,omitempty"`
-	GitTags             []GitTag             `bson:"git_tags,omitempty" json:"git_tags,omitempty"`
+
+	// GitTags stores tags that were pushed to this version, while TriggeredByGitTag is for versions created by tags
+	GitTags           []GitTag `bson:"git_tags,omitempty" json:"git_tags,omitempty"`
```

Comment:
I don't understand the difference between these two. Does one of them handle annotating a commit afterwards with a tag, and the other handle pushing both a commit and tag at once?

Reply:
Yeah I wasn't sure how to make it clear: 
GitTags denotes the version that a git tag was pushed to. This can be done whether or not a version is created.
If a git tag is pushed and also a version is supposed to be created for it, then we create a new version: this version doesn't _have_ the git tag because it doesn't specifically represent a commit like other waterfall commits do; rather, it was created _because of_ this git tag, which is the reason for TriggeredFromGitTag.

Reply:
Alternatively GitTag could have another field "TriggeredVersion": bool, and if the version was triggered from that tag then that bool could be set to true.